File: examples/sdk-java/src/main/java/com/crosscert/passkey/sdk/exception/PasskeyRateLimitException.java
package com.crosscert.passkey.sdk.exception;

import com.crosscert.passkey.sdk.envelope.EnvelopeFieldError;

import java.util.List;

public class PasskeyRateLimitException extends PasskeyApiException {
    private final long retryAfterSeconds;

    public PasskeyRateLimitException(String code, String message, String traceId,
                                     List<EnvelopeFieldError> fieldErrors,
                                     long retryAfterSeconds) {
        super(429, code, message, traceId, fieldErrors);
        this.retryAfterSeconds = retryAfterSeconds;
    }

    public long retryAfterSeconds() { return retryAfterSeconds; }
}

File: sdk-java/src/main/kotlin/com/crosscert/passkey/sdk/exception/PasskeyRateLimitException.kt
package com.crosscert.passkey.sdk.exception

import com.crosscert.passkey.sdk.envelope.EnvelopeFieldError

class PasskeyRateLimitException(
    code: String?,
    message: String?,
    traceId: String?,
    fieldErrors: List<EnvelopeFieldError>?,
    @get:JvmName("retryAfterSeconds") val retryAfterSeconds: Long,
) : PasskeyApiException(429, code, message, traceId, fieldErrors)
